src/main/python/org/broadinstitute/hellbender/vqsr_cnn/vqsr_cnn/models.py

```diff
@@ -0,0 +1,407 @@
+    Assert that the serialized tensor map and the recreated one are the same.
+
+    Arguments:
+        args.tensor_name: String which indicates tensor map to use or None
+        args.window_size: sites included in the tensor map
+        args.read_limit: Maximum reads included in the tensor map
+        args.annotations: List of annotations or None
+        semantics_json: Semantics json file (created with serialize_model_semantics())
+
+    Returns:
+        The Keras model
+    '''
+    with open(semantics_json, 'r') as infile:
+        semantics = json.load(infile)
+
+    if 'input_tensor_map' in semantics:
+        args.tensor_name = semantics['input_tensor_map_name']
+        args.window_size = semantics['window_size']
+        args.read_limit = semantics['read_limit']
+        tm = tensor_maps.get_tensor_channel_map_from_args(args)
+        assert(len(tm) == len(semantics['input_tensor_map']))
+        for key in tm:
+            assert(tm[key] == semantics['input_tensor_map'][key])
+
+    if 'input_annotations' in semantics:
+        args.annotations = semantics['input_annotations']
+
+    args.input_symbols = semantics['input_symbols']
+    args.labels = semantics['output_labels']
+
+    if 'data_dir' in semantics:
+        args.data_dir = semantics['data_dir']
+
+    weight_path_hd5 = os.path.join(os.path.dirname(semantics_json), semantics['architecture'])
+    print('Loading keras weight file from:', weight_path_hd5)
+    model = load_model(weight_path_hd5, custom_objects=get_metric_dict(args.labels))
+    model.summary()
+    return model
+
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# ~~~~~~~ Models ~~~~~~~~~~~~~~~~~~
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+def build_reference_annotation_model(args):
+    '''Build Reference 1d CNN model for classifying variants with skip connected annotations.
+
+    Convolutions followed by dense connection, concatenated with annotations.
+    Dynamically sets input channels based on args via tensor_maps.get_tensor_channel_map_from_args(args)
+    Uses the functional API.
+    Prints out model summary.
+
+    Arguments
+        args.tensor_name: The name of the tensor mapping which data goes to which channels
+        args.annotation_set: The variant annotation set, perhaps from a HaplotypeCaller VCF.
+        args.labels: The output labels (e.g. SNP, NOT_SNP, INDEL, NOT_INDEL)
+
+    Returns
+        The keras model
+    '''
+    if K.image_data_format() == 'channels_last':
+        channel_axis = -1
+    else:
+        channel_axis = 1
+
+    channel_map = tensor_maps.get_tensor_channel_map_from_args(args)
+    reference = Input(shape=(args.window_size, len(channel_map)), name=args.tensor_name)
+    conv_width = 12
+    conv_dropout = 0.1
+    fc_dropout = 0.2
+    x = Conv1D(filters=256, kernel_size=conv_width, activation="relu", kernel_initializer='he_normal')(reference)
+    x = Conv1D(filters=256, kernel_size=conv_width, activation="relu", kernel_initializer='he_normal')(x)
```

Comment:
did you try putting normalizations here too?

Reply:
Yes, and most of the time ROC performance decreases.  It is one of the covariates for the Bayesian Hyperparameter optimizer too and it doesn't recommends it either.  I'm a bit suspicious about this because with images Batch Normalization after every convolution seems to consistently give better results and faster training times...